missing macos arm64 (Apple Silicon) architecture support

# Problem

Trying to run the [examples](https://github.com/51Degrees/device-detection-java-examples) via `mvn clean install` on an arm64 macos machine fails with this record in the logs:

```
...
Caused by: java.lang.UnsatisfiedLinkError: The native library failed to load. This may mean that either the library itself or one of its dependencies is not accessible or not in the correct format for the target OS or architecture. A common missing library is libatomic1 on Linux. If this is the case, use your Linux package manager to install it on your system. If this is not possible, you may be able to use a different library by rebuilding the binary on the target system using the source code and instructions available on https://github.com/51degrees/device-detection-java. Inner error: Could not find the resource 'DeviceDetectionHashEngine-mac-aarch64.dylib'. Check the resource exists in the 'DeviceDetectionHashEngine' package or is present in the src/main/resources folder.
	at fiftyone.devicedetection.hash.engine.onpremise.interop.swig.DeviceDetectionHashEngineModuleJNI.<clinit>(DeviceDetectionHashEngineModuleJNI.java:287)
	at fiftyone.devicedetection.hash.engine.onpremise.interop.swig.ConfigHashSwig.<init>(ConfigHashSwig.java:40)
	at fiftyone.devicedetection.hash.engine.onpremise.flowelements.DeviceDetectionHashEngineBuilder.<init>(DeviceDetectionHashEngineBuilder.java:67)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	... 47 common frames omitted
```
The same command would succeed on an x64 macos machine.  The .jar package indeed does not contain the binary for the arm64 arch, but contains one for the x64: 

```sh
% unzip -t device-detection.hash.engine.on-premise-4.4.47.jar | grep dylib
    testing: DeviceDetectionHashEngine-mac-x64.dylib   OK
```

# Solution

Need to cross compile the arm64 binary on the macos-latest runner (that is of x64 architecture) - it is capable to compile the binary but will require skipping the tests for the arm64 architecture.

# Testing

The examples build and run on all platforms including macos arm64 with the above command.

--- a/.github/workflows/nightly-publish-main.yml
+++ b/.github/workflows/nightly-publish-main.yml
@@ -12,6 +12,7 @@ jobs:
     uses: 51Degrees/common-ci/.github/workflows/nightly-publish-main.yml@main
     with:
       repo-name: ${{ github.event.repository.name }}
+      org-name: ${{ github.event.repository.owner.login }}
       build-platform: ubuntu-20.04
     secrets: 
       token: ${{ secrets.ACCESS_TOKEN }}

--- a/device-detection.hash.engine.on-premise/src/main/cxx/CMakeLists.txt
+++ b/device-detection.hash.engine.on-premise/src/main/cxx/CMakeLists.txt
@@ -27,7 +27,9 @@ if (MSVC)
     endif()
 
 else()
-    if (${CMAKE_SYSTEM_PROCESSOR} MATCHES ^armhf)
+    if (DEFINED ARCH)
+        # Use the defined ARCH
+    elseif (${CMAKE_SYSTEM_PROCESSOR} MATCHES ^armhf)
         set(ARCH armhf)
     elseif (${CMAKE_SYSTEM_PROCESSOR} MATCHES ^aarch64)
         set(ARCH aarch64)

--- a/device-detection.hash.engine.on-premise/src/main/cxx/PreBuild.sh
+++ b/device-detection.hash.engine.on-premise/src/main/cxx/PreBuild.sh
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -eu
 
 # mvn clean will delete target so that's the place to be building
 mkdir -p target/cxx-build
@@ -10,3 +11,11 @@ cmake ../../src/main/cxx
 # put output in target/classes so that it will get jarred up
 echo "cmake build"
 cmake --build . --target fiftyone-hash-java
+
+if [[ $OSTYPE == darwin* ]]; then
+	echo "additional ARM64 build"
+	echo "configure"
+	cmake ../../src/main/cxx -DARCH=aarch64 -DCMAKE_OSX_ARCHITECTURES=arm64 -DBUILD_TESTING=OFF
+	echo "cmake build"
+	cmake --build . --target fiftyone-hash-java
+fi

